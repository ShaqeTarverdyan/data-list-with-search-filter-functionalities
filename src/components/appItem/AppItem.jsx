import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const ItemWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 1rem;
  min-height: 12rem;
  background-color: var(--color-white);
  box-shadow: 0 2px 3px 0 #e8ebed, 0 0 3px 0 #e8ebed;
  position: relative;
  width: 100%;
  margin-bottom: 1rem;

  &:hover {
    background-color: var(--color-gray);
    cursor: pointer;
  }
`;
const Title = styled.h3`
  color: var(--color-teal);
  font-size: 2rem;
  font-weight: 700;
`;
const Description = styled.p`
  color: var(--color-grayDark);
`;
const LeftSide = styled.div``;
const Top = styled.div`
  display: flex;
  align-items: top;
  justify-content: space-between;
`;

const CategoryItem = styled.div``;
const Categories = styled.div``;
const Subscriptions = styled.div`
  display: flex;
`;
const Subscription = styled.div`
  display: flex;
  margin-right: 1rem;
`;
const SubscriptionName = styled.div`
  margin-right: 0.5rem;
`;
const SubscriptionPrice = styled.div``;

const AppItem = ({ app }) => {
  return (
    <ItemWrapper data-testid="app-item">
      <Top>
        <LeftSide>
          <Title data-testid="name">{app.name}</Title>
          <Description data-testid="description">{app.description}</Description>
        </LeftSide>
        <Categories>
          {app.categories.map((category, index) => {
            return (
              <CategoryItem key={category + index}>{category}</CategoryItem>
            );
          })}
        </Categories>
      </Top>
      <Subscriptions>
        {app.subscriptions.map(({ name, price }, index) => {
          return (
            <Subscription key={index}>
              <SubscriptionName>{name}</SubscriptionName>
              <SubscriptionPrice>{price}</SubscriptionPrice>
            </Subscription>
          );
        })}
      </Subscriptions>
    </ItemWrapper>
  );
};

export default AppItem;

AppItem.propTypes = {
  app: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    categories: PropTypes.array,
    subscriptions: PropTypes.array,
  }),
};
