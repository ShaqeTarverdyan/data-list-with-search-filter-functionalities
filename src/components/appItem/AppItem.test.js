import React from "react";
import AppItem from "./AppItem";

import { render, screen } from "@testing-library/react";

const app = {
  id: "9b565b11-7311-5b5e-a699-97873dffb361",
  name: "Voice Report",
  description: "Calls reporting and analytics of your calls.",
  categories: ["Voice Analytics", "Reporting", "Optimization"],
  subscriptions: [
    {
      name: "Trial",
      price: 0,
    },
    {
      name: "Professional",
      price: 3500,
    },
  ],
};

describe("AppItem", () => {
  test("renders AppItem component", () => {
    render(<AppItem app={app} />);
    screen.debug();
    expect(screen.getByTestId("app-item"));
  });
});
