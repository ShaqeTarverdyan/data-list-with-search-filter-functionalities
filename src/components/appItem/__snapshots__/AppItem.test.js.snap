// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL imxVRw"
  >
    <div
      class="sc-eCImPb jduaVz"
    >
      <div
        class="sc-hKwDye"
      >
        <h3
          class="sc-gsDKAQ kbyjGO"
        >
          Voice Report
        </h3>
        <p
          class="sc-dkPtRN jqpBkB"
        >
          Calls reporting and analytics of your calls.
        </p>
      </div>
      <div
        class="sc-gKclnd"
      >
        <div
          class="sc-jRQBWg"
        >
          Voice Analytics
        </div>
        <div
          class="sc-jRQBWg"
        >
          Reporting
        </div>
        <div
          class="sc-jRQBWg"
        >
          Optimization
        </div>
      </div>
    </div>
    <div
      class="sc-iCfMLu gZJfOz"
    >
      <div
        class="sc-furwcr lhlxOv"
      >
        <div
          class="sc-pVTFL etPZCM"
        >
          Trial
        </div>
        <div
          class="sc-jrQzAO"
        >
          0
        </div>
      </div>
      <div
        class="sc-furwcr lhlxOv"
      >
        <div
          class="sc-pVTFL etPZCM"
        >
          Professional
        </div>
        <div
          class="sc-jrQzAO"
        >
          3500
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`testing the render of the AppItemcomponent 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL imxVRw"
  >
    <div
      class="sc-eCImPb jduaVz"
    >
      <div
        class="sc-hKwDye"
      >
        <h3
          class="sc-gsDKAQ kbyjGO"
        >
          Voice Report
        </h3>
        <p
          class="sc-dkPtRN jqpBkB"
        >
          Calls reporting and analytics of your calls.
        </p>
      </div>
      <div
        class="sc-gKclnd"
      >
        <div
          class="sc-jRQBWg"
        >
          Voice Analytics
        </div>
        <div
          class="sc-jRQBWg"
        >
          Reporting
        </div>
        <div
          class="sc-jRQBWg"
        >
          Optimization
        </div>
      </div>
    </div>
    <div
      class="sc-iCfMLu gZJfOz"
    >
      <div
        class="sc-furwcr lhlxOv"
      >
        <div
          class="sc-pVTFL etPZCM"
        >
          Trial
        </div>
        <div
          class="sc-jrQzAO"
        >
          0
        </div>
      </div>
      <div
        class="sc-furwcr lhlxOv"
      >
        <div
          class="sc-pVTFL etPZCM"
        >
          Professional
        </div>
        <div
          class="sc-jrQzAO"
        >
          3500
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
